data_science/archive_exploration/02 - exploring trees.ipynb

```diff
@@ -0,0 +1,353 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Trees\n",
+    "Archives are stored as trees - a node may represent a _piece_ (eg an individual bit of the archive subject's correspondence), an _item_ (typical leaf node in the tree, usually a box containing multiple pieces), a _section_, _series_, _subsection_, _subsubseries_, _subsubsubseries_ etc etc etc (describing the informational hierarchy though which context can be attributed to items and the collection as a whole can be understood, until, at the top of the tree, we reach the _collection-level_ record. The CALM data contains many collection level records, some of which are also contextually linked. All of these nodes in the trees are stored as individual rows in the CALM `.json`, and its position in the tree is described by the `AltRefNo` field. Let's open some up and have a look at their structure."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.core.display import display, HTML\n",
+    "\n",
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "sns.set_style('white')\n",
+    "plt.rcParams['figure.figsize'] = (30, 30)\n",
+    "\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import networkx as nx\n",
+    "\n",
+    "import re\n",
+    "\n",
+    "from umap import UMAP\n",
+    "from sklearn.decomposition import PCA\n",
+    "from sklearn.manifold import TSNE\n",
+    "\n",
+    "from tqdm import tqdm_notebook as tqdm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_json('data/calm_records.json')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df['AltRefNo']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `Level` field tells us whether the record is an item, a series, a subsubsection etc. However, I'm going to ignore this as the choice of when to use these hierarchy descriptors is somewhat arbitrary, chosen by the archivist in question to best fit the archive. I'm told that what _really_ matters is the _structure_ of the archive, which is all contained in the `AltRefNo`s.  \n",
```

Comment:
An example of the AltRefNo, and how it indicates structure, might serve as a useful reference here.